### DISCUSSION TITLE
Failed to resolve versions ethers_solc

### DISCUSSION THREAD

**Discussion by MeetEatMeat** [foundry-rs/foundry#2232] [Help]
I'm trying to build a project which has several pragma versions in it's dependencies list, but then i'm getting an error below:
<img width="761" alt="Снимок экрана 2022-07-07 в 17 16 43" src="https://user-images.githubusercontent.com/34601776/177783386-2e0d4b71-0631-4e3e-a09e-f8183816dba8.png">
 How to resolve it?

**Comment by onbjerg** [comment]
AlgebraPool_flattened.sol only works on solc 0.7.6, but most of your other contracts (e.g. OpenZeppelin) requires solc 0.8.0 or above. You need to change the pragmas so all the versions are compatible.

**Reply by FaybianB** [reply]
Is there any way to disable or bypass this error? It can be a bit of a hassle to resolve these pragma differences, especially in the case of importing files from libraries outside of our codebase.

**Reply by mds1** [reply]
There is not a way to bypass this error—this is a solc limitation, because during compilation a given contract can of course only be compiled with a single version, so the specified versions must be compatible

**Reply by 0xy0ung** [reply]
hey, I facing same issue now. When I imported a outside lib, and I can't change the pragmas because the project is using 3 different solc-version. I have another way to solve it right now? Thanks in advance!
